flink-table-store-core/src/main/java/org/apache/flink/table/store/file/data/DataFilePathFactory.java

```diff
@@ -34,20 +34,30 @@ public class DataFilePathFactory {
     private final String uuid;
 
     private final AtomicInteger pathCount;
+    private final String formatIdentifier;
 
     public DataFilePathFactory(Path root, String partition, int bucket) {
```

Comment:
I think we can do without this constructor, the `formatIdentifier` is not a nullable field, and the following can be done without having the special treatment of null.

Reply:
Fair enough, deleted.